warehouse/query-core/src/main/java/datawave/audit/DatawaveSelectorExtractor.java

```diff
@@ -22,13 +22,14 @@
         ASTJexlScript jexlScript = null;
         
         try {
-            jexlScript = JexlASTHelper.parseJexlQuery(query.getQuery());
+            // Parse & Flatten to reduce the number of node traversals in later method calls
+            jexlScript = JexlASTHelper.parseAndFlattenJexlQuery(query.getQuery());
```

Comment:
You need to remove the call to the TreeFlatteningRebuildingVisitor.flatten below on line 39

Reply:
Done.